src/components/index.esm.js

```diff
@@ -41,11 +41,18 @@ export * from './pagination-nav'
 export * from './popover'
 export * from './progress'
 export * from './spinner'
-export * from './table'
+// Webpack 4 has difficulties with re-eport of re-export optimizations
+// So we import the table components indivisulaly here for better tree shaking.
+// the table components are one of hte largest components in the library
+// so we do this here to help with consumer optimizations.
+// Webpack v5 fixes the optimizations with re-export of re-exports so this
+// can be reverted back to `export * from './table'` when Webpack v5 is released.
+// https://github.com/webpack/webpack/pull/9203
+// export * from './table'
+export { default as BTable } from './table/table'
+export { default as BTableLite } from './table/table-lite'
```

Comment:
Maybe we should use this syntax for all components when webpack has difficulties with the other one.

Reply:
I thought about that...  but most component dirs have only one component (i.e. embed, jubotron, etc) or a bunch of smaller components (layout, dropdown) that would always be used together (or have insignificant size).

Then I thought of maintenance... we would need to maintain the imports (or automate the creation of the index file), when Webpack5 rolls out (they have https://www.npmjs.com/package/webpack/v/5.0.0-alpha.15 released at the moment) we would need to revert back.

What happens is that webpack will handle `export * from 'a'`, but doesn't properly track if `'a'` uses `export * from 'b'` (which is fixed in v5)

Reply:
What webpack does (in some cases) is import  `src/components/table/index` into the final bundle, when all one imported was `BTableLite`.

The only other places where there would be a bit of savings would be with image (users may not use img-lazy)

And this is only applicable when importing individual components from a component group. Importing plugins has no issues.

Reply:
Here is a breakdown of imported modules:

![image](https://user-images.githubusercontent.com/2781561/59148362-657dd080-89de-11e9-8d19-f9039ca9e0c9.png)

On the upper right is table (with the helper mixins and utils). it is bigger than modal.

Reply:
But maybe we could do the re-export individually, until Webpack v5.0.0 is released.

Reply:
I did find that

```js
export * from './table'
```
followed by
```js
export { default as BTable } from './table/table'
export { default as BTableLite } from './table/table-lite'
```

Worked well, and webpack and rollup correctly handled the duplicate export names

Reply:
Although it really only benefits table/table-lite

Reply:
But after a bit of thought, by exporting each component individually (for not util Webpack 5 or Webpack 4 is improved), does prevent the plugin code from being imported when not importing plugins.